### PR TITLE
Update coding-standard to major version 3.0.2

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
1) Update coding-standard to major version 3. That gets php-cs-fixer major version 3 also.

2) Apply code changes suggested by php-cs-fixer v3. IMO it now applies PSR12 by default, and other breaking changes to code style stuff.

I have not added a changelog, because this is a code-style-only change. There is not observable behavior change at any level (unit test, acceptance test, or production). Do we write a changelog anyway for this sort of thing?

## Related Issue
- part of https://github.com/owncloud/coding-standard/issues/22

## Motivation and Context
Have beautiful code :)

## How Has This Been Tested?
CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Comment by phil-davis** [comment]
PR #38824 has 1130 files changed, +16,111 -10,697 lines

This PR removes the forced-removal of leading `\` from functions like `\preg_match`.

Now there are "only" 605 files changed, +9,560 -4,146 lines

That is significantly less diffs.

**Comment by JanAckermann** [comment]
:rocket: Makes the code much more beautiful!

File: apps/comments/lib/Activity/Extension.php [hunk 1]
```diff
@@ -36,10 +36,10 @@
  * @package OCA\Comments\Activity
  */
 class Extension implements IExtension {
-	const APP_NAME = 'comments';
+	public const APP_NAME = 'comments';
```
Comment (phil-davis) [inline apps/comments/lib/Activity/Extension.php:39]: class constants are public by default. This change makes that explicit. This seems to be "a good thing".

We could look through to find class constants that are only ever referenced inside the class, and make them private?
(but that would be a separate task - the change here is completely backward-compatible and no risk)

File: apps/comments/lib/Activity/Listener.php [hunk 2]
```diff
@@ -52,11 +52,13 @@ class Listener {
...
-								IUserSession $session,
-								IAppManager $appManager,
-								IMountProviderCollection $mountCollection,
-								IRootFolder $rootFolder) {
+	public function __construct(
+		IManager $activityManager,
+		IUserSession $session,
+		IAppManager $appManager,
+		IMountProviderCollection $mountCollection,
+		IRootFolder $rootFolder
```
Comment (phil-davis) [inline apps/comments/lib/Activity/Listener.php:60]: Now it puts the parameters all on separate lines. The change here looks like "a good thing"

**Review by AlexAndBear — APPROVED** [review]

**Comment by JanAckermann** [comment]
SMELLS Good to me :rocket:

**Review by JammingBen — APPROVED** [review]
Awesome, really makes the code look cleaner 👍

**Pull request merged**